src/main/java/org/generationcp/middleware/service/impl/study/StudyServiceImpl.java

```diff
@@ -616,6 +617,11 @@ public boolean studyHasGivenDatasetType(final Integer studyId, final Integer dat
 		return (!org.springframework.util.CollectionUtils.isEmpty(datasets));
 	}
 
+	@Override
+	public boolean environmentHasGivenDatasetType(final Integer environmentId, final DatasetTypeEnum datasetTypeId) {
+		return Util.getIntValue(this.studyDataManager.getDatasetIdByEnvironmentIdAndDatasetType(environmentId, datasetTypeId)) > 0;
```

Comment:
Let's remove getDatasetIdByEnvironmentIdAndDatasetType from StudyDataManager and move to StudyInstanceService as `getDatasetIdByInstanceIdAndDatasetType `. Its previous usage (StudyResourceBrapi) can use this